index/scorch/segment/mem/posting.go

```diff
@@ -155,6 +155,24 @@ func (i *PostingsIterator) Next() (segment.Posting, error) {
 	return &i.reuse, nil
 }
 
+func (i *PostingsIterator) Advance(localDocNumber uint64) (segment.Posting, error) {
```

Comment:
one thought that pops up is whether Advance() would ever be called to "advance" to the same localDocNumber that it's already on?  If so, then that might not work with this implementation?

Reply:
whoops..I missed it..Done! thanks!

Reply:
One thought on the variable naming.  When looking at code at the segment level (api or implementation) we should just use "doc number" not "local doc number".  The reason is that in the context of a segment, there is no distinction, it's just the number of a document inside the segment.